andela/ah-infinity-stones
# #161966915 User Profiles

#### What does this PR do?
- sets up jwt tokens
- uses sendgrid for email verification
- extends user profiles

#### Description of Task to be completed?
> Setup postgres test database



#### How should this be manually tested?
`$ git clone -b ch-test-delete-update-notifications2-161966897 https://github.com/andela/ah-infinity-stones` then
`$ cd ah-infinity-stones`
> Create and activate virtual env
virtualenv -p python3 venv
`$ source venv/bin/activate`
`$ pip3 install -r requirements.txt to install the dependencies`
> Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice:
> Register user via `$ python3 manage.py createsuperuser --username john --email john.doe@example.com`
> Check Inbox
> Login and access restricted endpoints
> POST http://127.0.0.1:8000/api/users/login/
> GET http://127.0.0.1:8000/api/profiles/
> GET http://127.0.0.1:8000/api/profiles/?page=2
> GET http://127.0.0.1:8000/api/profiles/u
> GET PUT http://127.0.0.1:8000/api/profiles/<username>

```
export DATABASE_USER='user'
export DATABASE_PASSWORD='xxx'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'

 export EMAIL_HOST_USER='sendgrid_username'
 export EMAIL_HOST_PASSWORD='sendgrid_pswd'
 
```
> Activate the env variables by `$source .env`
> Run migrations: $ python3 manage.py migrate
> Run tests - $ python3 manage.py test

#### What are the relevant pivotal tracker stories?
[#161966915](https://www.pivotaltracker.com/story/show/161966915)

#### Screenshots
<img width="1235" alt="screen shot 2018-12-11 at 16 32 31" src="https://user-images.githubusercontent.com/39657549/49935957-8b209600-fee3-11e8-9926-1b514cdafe47.png">
<img width="1280" alt="screen shot 2018-12-13 at 14 28 56" src="https://user-images.githubusercontent.com/39657549/49935958-8bb92c80-fee3-11e8-8929-da21dcac8093.png">
<img width="1280" alt="screen shot 2018-12-13 at 14 29 04" src="https://user-images.githubusercontent.com/39657549/49935959-8bb92c80-fee3-11e8-8d5e-1ecc27da8d31.png">
<img width="1280" alt="screen shot 2018-12-13 at 14 29 09" src="https://user-images.githubusercontent.com/39657549/49935960-8bb92c80-fee3-11e8-8f38-213e434eb442.png">
<img width="1280" alt="screen shot 2018-12-13 at 14 29 13" src="https://user-images.githubusercontent.com/39657549/49935961-8c51c300-fee3-11e8-89f4-3989a89843dd.png">
